src/main/kotlin/apply/application/MissionService.kt

```diff
@@ -58,15 +61,34 @@ class MissionService(
         return MissionData(mission, evaluation)
     }
 
-    fun findAllByRecruitmentId(recruitmentId: Long): List<MissionResponse> {
+    fun findAllByRecruitmentId(recruitmentId: Long): List<MissionAndEvaluationResponse> {
         val evaluationsById = evaluationRepository.findAllByRecruitmentId(recruitmentId).associateBy { it.id }
         val missions = missionRepository.findAllByEvaluationIdIn(evaluationsById.keys)
-        return missions.map { MissionResponse(it, evaluationsById.getValue(it.evaluationId)) }
+        return missions.map { MissionAndEvaluationResponse(it, evaluationsById.getValue(it.evaluationId)) }
     }
 
     fun deleteById(id: Long) {
         val mission = missionRepository.getById(id)
         check(!mission.submittable) { "제출 가능한 과제는 삭제할 수 없습니다." }
         missionRepository.deleteById(id)
     }
+
+    fun findAllByUserIdAndRecruitmentId(userId: Long, recruitmentId: Long): List<MissionResponse> {
+        val evaluationIds = evaluationRepository.findAllByRecruitmentId(recruitmentId).map { it.id }
+        val includedEvaluationIds = evaluationIds.filter {
+            evaluationTargetRepository.existsByUserIdAndEvaluationId(userId, it)
+        }
+        return missionRepository.findAllByEvaluationIdIn(includedEvaluationIds).map {
+            MissionResponse(
+                it.id,
+                it.title,
+                it.description,
+                it.submittable,
+                assignmentRepository.existsByUserIdAndMissionId(userId, it.id),
+                it.period.startDateTime,
+                it.period.endDateTime,
+                it.status
+            )
+        }
```

Comment:
현재 코드와 아래 코드에서 생성되는 쿼리의 차이점은 무엇일까요?

```suggestion
        val assignments = assignmentRepository.findAllByUserId(userId)
        return missionRepository.findAllByEvaluationIdIn(includedEvaluationIds)
            .map { mission -> MissionResponse(mission, assignments.any { it.missionId == mission.id }) }
```

Reply:
아 이렇게 하면 매 과제 마다 쿼리를 던져야 하는걸 1번으로 줄일 수 있겠네요!
고민되던 부분이었는데 감사합니다! ㅠㅠ